src/pyuvdata/uvbeam/uvbeam.py

```diff
@@ -1340,6 +1341,10 @@ def _interp_az_za_rect_spline(
         check_azza_domain : bool
             Whether to check the domain of az/za to ensure that they are covered by the
             intrinsic data array. Checking them can be quite computationally expensive.
+        spatial_interp_func : str
+            The spatial interpolation function to use. Options are 'RectBivariateSpline'
+            and 'map_coordinates'. If 'map_coordinates' is chosen, the input data must
+            be on a regular grid.
```

Comment:
I believe that when the pixel coordinate system is `"az_za"` (which is required for this method), the beam is required to be on a regular grid. I thought there was something enforcing this in the check, but I just looked and it's only enforced in the `read_cst` method. There's also a warning (but not an error) in the `select` method if you do a select that creates an irregular grid. I guess this is something we should talk more about, but the original intent, at least, was that these beams would have regularly gridded coordinates.

Reply:
We talked about this on the telecon yesterday and @steven-murray suggested we add the regularity checks in this PR. He said he could do it if you want, I'm also happy to help.